Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/StereoPicture/EyeLink_StereoPicture.m

```diff
@@ -0,0 +1,408 @@
+                transferFile; % See transferFile function below
+                cleanup; % Abort experiment (see cleanup function below)
+                return
+            end
+            % End trial if spacebar is pressed
+            [~, RtEnd, keyCode] = KbCheck;
+            if keyCode(spaceBar)
+                % Write message to EDF file to mark the spacebar press time
+                Eyelink('Message', 'KEY_PRESSED');
+                reactionTime = round((RtEnd-RtStart)*1000); % Calculate RT from stimulus onset
+                break;
+            end
+        end % End of while loop
+        
+        % Draw blank screen at end of trial
+        for itScr = 0:1
+            Screen('SelectStereoDrawBuffer', window, itScr);
+            Screen('FillRect', window, el.backgroundcolour); % Prepare grey background on backbuffer
+%            Screen('DrawTexture', window, backgroundTexture(itScr+1)); % Prepare background texture on backbuffer
+        end
+        Screen('Flip', window); % Present blank screen
+        % Write message to EDF file to mark time when blank screen is presented
+        Eyelink('Message', 'BLANK_SCREEN');
+        % Write !V CLEAR message to EDF file: creates blank backdrop for DataViewer
+        % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Simple Drawing
+        Eyelink('Message', '!V CLEAR %d %d %d', el.backgroundcolour(1), el.backgroundcolour(2), el.backgroundcolour(3));
+        
+        % Stop recording eye movements at the end of each trial
+        WaitSecs(0.1); % Add 100 msec of data to catch final events before stopping
+        Eyelink('StopRecording'); % Stop tracker recording
+        WaitSecs(0.001); % Allow some time for recording to stop
+        
+        % STEP 5.5: CREATE VARIABLES FOR DATAVIEWER; END TRIAL
+        
+        % Write !V TRIAL_VAR messages to EDF file: creates trial variables in DataViewer
+        % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Trial Message Commands
+        Eyelink('Message', '!V TRIAL_VAR iteration %d', i); % Trial iteration
+        Eyelink('Message', '!V TRIAL_VAR leftImage %s', imgNameLeft); % Image name
+        Eyelink('Message', '!V TRIAL_VAR rightImage %s', imgNameRight); % Image name
+        WaitSecs(0.001); % Allow some time between messages. Some messages can be lost if too many are written at the same time
+        Eyelink('Message', '!V TRIAL_VAR rt %d', reactionTime); % Reaction time
+        % Write TRIAL_RESULT message to EDF file: marks the end of a trial for DataViewer
+        % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Defining the Start and End of a Trial
+        Eyelink('Message', 'TRIAL_RESULT 0');
+        WaitSecs(0.01); % Allow some time before ending the trial
+        
+        % Clear Screen() textures that were initialized for each trial iteration
+        for itScr = 0:1

```

Comment:
No loop needed. You can pass in imgTexture as a whole - it will close all textures in that vector/array.

Reply:
understood and agreed the StereoMode demo needs to be revamped to reflect the most current PTB conventions. I will include this as a priority item in the next round of revisions following completion of the rest of this pull request to allow for additional time to make those changes and thorough testing before publishing